src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleLogFormatter.cs

```diff
@@ -0,0 +1,168 @@
+                    writer.WriteString("message", exception.Message.ToString());
+                    writer.WriteString("type", exception.GetType().ToString());
+                    writer.WriteStartArray("stackTrace");
+                    if (exception?.StackTrace != null)
+                    {
+                        foreach (var stackTraceLines in exception?.StackTrace?.Split(Environment.NewLine))
+                        {
+                            JsonSerializer.Serialize<string>(writer, stackTraceLines);
+                        }
+                    }
+                    writer.WriteEndArray();
+                    writer.WriteNumber("hResult", exception.HResult);
+                    writer.WriteEndObject();
+                }
+
+                GetScopeInformation(writer, scopeProvider);
+
+                writer.WriteEndObject();
+
+                writer.Flush();
+            }
+            textWriter.Write(Encoding.UTF8.GetString(output.WrittenMemory.Span));
+            textWriter.Write(Environment.NewLine);
+        }
+
+
+        private static string GetLogLevelString(LogLevel logLevel)
+        {
+            switch (logLevel)
+            {
+                case LogLevel.Trace:
+                    return "json_trace";
+                case LogLevel.Debug:
+                    return "json_debug";
+                case LogLevel.Information:
+                    return "json_info";
+                case LogLevel.Warning:
+                    return "json_warn";
+                case LogLevel.Error:
+                    return "json_fail";
+                case LogLevel.Critical:
+                    return "json_critical";
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(logLevel));
+            }
+        }
+
+        private void GetScopeInformation(Utf8JsonWriter writer, IExternalScopeProvider scopeProvider)
+        {
+            try
+            {
+                if (FormatterOptions.IncludeScopes && scopeProvider != null)
+                {
+                    writer.WriteStartObject("scopes");
+                    scopeProvider.ForEachScope((scope, state) =>
+                    {
+                        if (scope is IReadOnlyList<KeyValuePair<string, object>> kvps)
+                        {
+                            foreach (var kvp in kvps)
+                            {
+                                if (kvp.Value is string ss)
+                                    state.WriteString(kvp.Key, ss);
+                                else
+                                if (kvp.Value is int ii)
+                                    state.WriteNumber(kvp.Key, ii);
+                                else
+                                {
+                                    // check how this work
+                                    state.WritePropertyName(kvp.Key);
+                                    JsonSerializer.Serialize(state, kvp.Value);
```

Comment:
What do other loggers do here?  Should we really be serializing these or should we `.ToString()` them and write as strings?

Reply:
Other formatters don't go this far to check if the scope is a key value pair:

![image](https://user-images.githubusercontent.com/5897654/86858360-5cde4a80-c075-11ea-84f1-6c26b96152f1.png)

Sample
```c#
                using (_logger.BeginScope("This is a scope message with number: {CustomNumber}", 11123))
                {
                    _logger.LogInformation(exception, "exception message with {0}", "stacktrace");
                }
```

Reply:
Hmmmm, good question. I think we should be using `ToString()`. Nested objects aren't really exposed like this.

Reply:
OK then so it would change to this:

![image](https://user-images.githubusercontent.com/5897654/86859825-61f0c900-c078-11ea-8bcb-a9e168866497.png)